core/src/main/java/io/questdb/cairo/O3PurgeDiscoveryJob.java

```diff
@@ -189,22 +189,20 @@ private static void processDir(
     @Override
     protected boolean doRun(int workerId, long cursor) {
         final O3PurgeDiscoveryTask task = queue.get(cursor);
-        try {
-            return discoverPartitions(
-                    configuration.getFilesFacade(),
-                    sink[workerId],
-                    nativeLPSZ[workerId],
-                    txnList[workerId],
-                    purgeQueue,
-                    purgePubSeq,
-                    configuration.getRoot(),
-                    task.getTableName(),
-                    task.getPartitionBy(),
-                    task.getTimestamp(),
-                    task.getTxnScoreboard()
-            );
-        } finally {
-            subSeq.done(cursor);
-        }
+        final boolean useful = discoverPartitions(
```

Comment:
Why is it better than try finally? You don't want to release sequence in case of exception?

Reply:
This is because exception is never thrown from that code block. Eg there is no code to throw exception. 

Having finally on other hand reduces test coverage. Tools highlight finally block that is never triggered by exception